extjsdk/src/main/java/io/vantiq/extjsdk/ExtensionWebSocketListener.java

```diff
@@ -323,8 +335,11 @@ else if (msg.get("op") == null) {
             }
             else {
                 if ((int) msg.get("status") == 200 && !client.isAuthed()) {
-                    // TODO should this be completed on all receipts or only for success?
-                    client.authFuture.complete(true);
+                    // Forcibly setting in case an error occurred before succeeding
+                    client.authFuture.obtrudeValue(true);
```

Comment:
Doesn't this return that things worked regardless of if they did?

Reply:
This and the next one were intentional. What it does is force the value to be true even if it was false before, i.e. if it tries and succeeds after failing. That way anything that saved the value updates to anything that ever called it. However, I did just check and realize that it may cause issues with some queue-like functionality I implemented for websocket->authentication->connection, so I will go back and work on that bit.